src/CellData/SkeletonCellFieldPair.jl

```diff
@@ -0,0 +1,98 @@
+"""
+  SkeletonCellFieldPair is a special construct for allowing uh.plus
+  and uh.minus to be two different CellFields. In particular, it is
+  useful when we need one of the CellFields to be the dualized
+  version of the other for performing ForwardDiff AD of a Skeleton
+  integration DomainContribution wrt to the degrees of freedom
+  of the CellField, plus and minus sensitivities done separately,
+  so as to restrict the interaction between the dual numbers.
+
+  It takes in two CellFields and stores plus version of CellFieldAt
+  for the first CellField and minus version of CellFieldAt for the
+  second the CellField. SkeletonCellFieldPair is associated with the
+  same triangulations as the CellFields
+"""
+struct SkeletonCellFieldPair{
+  P<:CellField, M<:CellField, T<:Triangulation} <: CellField
+
+  cf_plus::P
+  cf_minus::M
+  trian::T # Background Triangulation to which CellFields belong to
+
+  function SkeletonCellFieldPair(
+    cf_plus_plus::CellFieldAt,
+    cf_minus_minus::CellFieldAt,
+    trian::Triangulation)
+
+    P = typeof(cf_plus_plus)
+    M = typeof(cf_minus_minus)
+    T = typeof(trian)
+
+    new{P,M,T}(cf_plus_plus,cf_minus_minus,trian)
+  end
+end
+
+function SkeletonCellFieldPair(
```

Comment:
Why do we actually need this outer constructor? I would remove it, while putting part of the checks inside it to the inner constructor.

Reply:
In the second constructor for `SkeletonCellFieldPair`  we want to check if `dim(train) == dim(strain) + 1` so I kept `strian`  as an argument in the constructor

Reply:
Can we defer this check to a latter stage?

Reply:
Yeah it can be done at a later stage when the integration on the Skeleton terms happens but this will be repeated every single time the integration happens over iterations, which is not necessary I guess.

Reply:
> Yeah it can be done at a later stage when the integration on the Skeleton terms happens but this will be repeated every single time the integration happens over iterations, which is not necessary I guess.

I think it is better to defer this check when the integration on the Skeleton term happens. This is more aligned with how the Gridap high-level API is designed. Besides, there is no guarantee that the SkeletonTriangulation that you pass to the constructor is going to be the same as the one that you use in the weak formulation.

Reply:
ohk, so should I go ahead and remove the `strian` in the constructor and also remove the related check?

Reply:
> ohk, so should I go ahead and remove the strian in the constructor and also remove the related check?

yes, please. The check should be actually moved to another place, not 100% removed from the project.

Reply:
yeah I ll do it after the CI completes, to see if there are more errors. But isn't the check, or some other compatibility check automatically happening when we integrate the `SkeletonCellFieldPair` over Skeleton Triangulation?

Reply:
>  But isn't the check, or some other compatibility check automatically happening when we integrate the SkeletonCellFieldPair over Skeleton Triangulation?

Perhaps you are right. Can you try to find in the code where such a test is being performed?

Reply:
I ll try to dig into the code and check this and update..